Floofbot/Modules/ModMail.cs

```diff
@@ -0,0 +1,195 @@
+﻿using Discord;
+using Discord.Addons.Interactive;
+using Discord.Commands;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Floofbot.Modules
+{
+    [Summary("Send a message directly to the server's moderators")]
+    [Group("modmail")]
+    public class ModMailModule : InteractiveBase
+    {
+        private FloofDataContext _floofDb;
+        static readonly ulong RFURRY_SERVER_ID =225980129799700481; // TODO: Replace this so that it works on more servers
+        public ModMailModule(FloofDataContext _floofDB)
+        {
+            _floofDb = _floofDB;
+        }
+
+        [Command("")]
+        public async Task sendModMail([Summary("Message Content")][Remainder] string content)
+        {
+            try
+            {
+                // get values
+                var serverConfig = _floofDb.ModMails.Find(RFURRY_SERVER_ID);
+                IGuild guild = Context.Client.GetGuild(RFURRY_SERVER_ID); // can return null
+                Discord.ITextChannel channel = await guild.GetTextChannelAsync((ulong)serverConfig.ChannelId); // can return null
+                IRole role = null;
+                if (serverConfig.ModRoleId != null)
+                {
+                    role = guild.GetRole((ulong)serverConfig.ModRoleId); // can return null
+                }
+
+                if (serverConfig == null) // not configured
+                {
+                    return;
+                }
+
+                if (serverConfig.IsEnabled == false || guild == null || channel == null) // disabled OR channel for mails not set
+                {
+                    return;
+                }
+
+                if (string.IsNullOrEmpty(content))
```

Comment:
Content checking/command usage should probably be up top

Reply:
fixed